### ISSUE TITLE
Some memory cleanup methods required

### ISSUE THREAD

**Issue by hovercraft-github** [drom/node-iio#3]
In order to be able calling buf_read and scan_context related methods in a loop we need to destroy previously created objects, otherwise the memory leak will happen.

**Comment by drom** [comment]
Agree. We have to find a good ASYNC method for infinite, continues read / write on the device without memory leakage. Any ideas?
